rust/lance/src/index.rs

```diff
@@ -296,29 +297,47 @@ impl DatasetIndexExt for Dataset {
     }
 
     #[instrument(skip_all)]
-    async fn optimize_indices(&mut self) -> Result<()> {
+    async fn optimize_indices(&mut self, options: &OptimizeOptions) -> Result<()> {
         let dataset = Arc::new(self.clone());
         // Append index
         let indices = self.load_indices().await?;
 
+        let mut name_to_indices = HashMap::new();
+        for idx in indices.iter() {
+            name_to_indices
+                .entry(idx.name.clone())
+                .or_insert_with(Vec::new)
+                .push(idx);
+        }
```

Comment:
Nit: maybe prefer `HashMap::from_iter`?

There's unlikely to be very many entries here so it's probably not a real performance concern but it might be more readable / consistent with the code base.

Reply:
I did not success last time. This is a `HashMap<String, Vec<&Index>>`,  how can we do `from_iter()` from a `Iterator<String, &INdex>` to do a basically group by String.

I'd like to make it work with from_iter if possible.

Reply:
Sorry, I did not realize it is a group by.  I tried itertools groupby once but couldn't figure it out either.